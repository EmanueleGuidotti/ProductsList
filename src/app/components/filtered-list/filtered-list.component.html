<mat-grid-list cols="3" rowHeight="100" md-cols="3" md-gutter="1em">
    <mat-grid-tile *ngFor="let vehicle of vehicles">
        <mat-card class="vehicle-item">
            <img [src]="vehicle.img">
            <h4 class="type">{{vehicle.type}}</h4>
            <p class="brand">{{vehicle.brand}}</p>
        </mat-card>
    </mat-grid-tile>
</mat-grid-list>
