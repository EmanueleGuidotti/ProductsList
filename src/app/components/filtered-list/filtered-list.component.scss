.vehicle-item {
  display: flex;
  flex-direction: row;
  width: 90%;
  margin-left: 5%;
  margin-right: 5%;
  img {
    width: 30%;
    max-width: 30%;
    flex: 1;
  }
  .type, .brand {
    flex: 1;
    margin-left: 10px;
    color: white;
  }
}