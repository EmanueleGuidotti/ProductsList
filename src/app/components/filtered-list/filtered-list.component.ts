import {
    ChangeDetectionStrategy, ChangeDetectorRef,
    Component,
    Input, OnChanges,
    OnInit, SimpleChanges
} from '@angular/core';
import { DataModel } from '../../data.model';

@Component({
	selector: 'filtered-list-component',
	templateUrl: './filtered-list.component.html',
	styleUrls: ['./filtered-list.component.scss'],
	changeDetection: ChangeDetectionStrategy.OnPush
})
export class FilteredListComponent implements OnInit, OnChanges {
	@Input() vehicles: DataModel[];

	constructor(public cdf: ChangeDetectorRef) {}

	ngOnInit() {
    }

    ngOnChanges(changes: SimpleChanges): void {
    }
}
