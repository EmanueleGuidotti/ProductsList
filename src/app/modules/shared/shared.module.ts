import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { MatSelectModule, MatGridListModule } from '@angular/material';
import { MatListModule } from '@angular/material/list';
import {MatCardModule} from "@angular/material/card";

@NgModule({
	declarations: [],
	imports: [CommonModule, MatSelectModule, MatGridListModule, MatListModule, MatCardModule],
	exports: [MatSelectModule, MatGridListModule, MatListModule, MatCardModule],
	providers: []
})
export class SharedModule {}
