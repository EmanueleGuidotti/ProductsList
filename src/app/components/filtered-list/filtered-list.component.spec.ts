import {async, ComponentFixture, inject, TestBed} from '@angular/core/testing';

import { FilteredListComponent } from './filtered-list.component';
import { DebugElement } from '@angular/core';
import {BrowserAnimationsModule} from "@angular/platform-browser/animations";
import {SharedModule} from "../../modules/shared/shared.module";
import {DataModel} from "../../data.model";
import * as data from "../../../../mocks/data.json";
const redItems =  [
    {
        "brand": "Bugatti Veyron",
        "colors":  [
            "red",
            "black",
        ],
        "id": 1,
        "img": "https://upload.wikimedia.org/wikipedia/commons/thumb/c/c9/Bugatti_Veyron_16.4_%E2%80%93_Frontansicht_%281%29%2C_5._April_2012%2C_D%C3%BCsseldorf.jpg/520px-Bugatti_Veyron_16.4_%E2%80%93_Frontansicht_%281%29%2C_5._April_2012%2C_D%C3%BCsseldorf.jpg",
        "type": "car",
    },
    {
        "brand": "Boeing 787 Dreamliner",
        "colors":  [
            "red",
            "white",
            "black",
            "green",
        ],
        "id": 2,
        "img": "https://upload.wikimedia.org/wikipedia/commons/thumb/1/15/All_Nippon_Airways_Boeing_787-8_Dreamliner_JA801A_OKJ_in_flight.jpg/600px-All_Nippon_Airways_Boeing_787-8_Dreamliner_JA801A_OKJ_in_flight.jpg",
        "type": "airplane",
    },
    {
        "brand": "Canadair North Star",
        "colors":  [
            "red",
            "blue",
            "yellow",
            "green",
        ],
        "id": 4,
        "img": "https://upload.wikimedia.org/wikipedia/commons/thumb/1/1f/BOAC_C-4_Argonaut_Heathrow_1954.jpg/600px-BOAC_C-4_Argonaut_Heathrow_1954.jpg",
        "type": "airplane",
    },
    {
        "brand": "Airbus A400M Atlas",
        "colors":  [
            "red",
            "white",
        ],
        "id": 5,
        "img": "https://upload.wikimedia.org/wikipedia/commons/thumb/1/1a/A400M-1969.jpg/600px-A400M-1969.jpg",
        "type": "airplane",
    },
    {
        "brand": "Prairie 2-6-2",
        "colors":  [
            "red",
            "white",
            "grey",
        ],
        "id": 7,
        "img": "https://upload.wikimedia.org/wikipedia/commons/thumb/d/d2/CFR_Steam_locomotive.jpg/600px-CFR_Steam_locomotive.jpg",
        "type": "train",
    },
    {
        "brand": "Amer 4-4-0",
        "colors":  [
            "red",
            "black",
        ],
        "id": 9,
        "img": "https://upload.wikimedia.org/wikipedia/commons/thumb/1/1c/440woodcut.jpg/600px-440woodcut.jpg",
        "type": "train",
    },
    {
        "brand": "Ferrari F40",
        "colors":  [
            "red",
            "yellow",
        ],
        "id": 10,
        "img": "https://upload.wikimedia.org/wikipedia/commons/thumb/c/cb/F40_Ferrari_20090509.jpg/1920px-F40_Ferrari_20090509.jpg",
        "type": "car",
    },
]

describe('FilteredListComponent', () => {
	let component: FilteredListComponent;
	let fixture: ComponentFixture<FilteredListComponent>;
	let debugElement: DebugElement;

	beforeEach(async(() => {
		TestBed.configureTestingModule({
            imports: [
                BrowserAnimationsModule,
                SharedModule,
            ],
			declarations: [FilteredListComponent]
		}).compileComponents();
		fixture = TestBed.createComponent(FilteredListComponent);
		component = fixture.componentInstance;
		debugElement = fixture.debugElement;
	}));

	beforeEach(() => {
		fixture = TestBed.createComponent(FilteredListComponent);
		component = fixture.componentInstance;
		fixture.detectChanges();
	});

	it('should create', () => {
		expect(component).toBeTruthy();
	});

    it('should create the component', () => {
        const jsonData: DataModel[] = data.getAll;
        const listInstance = debugElement.componentInstance;
        component.vehicles = jsonData;
        component.cdf.markForCheck();
        fixture.autoDetectChanges(true);
        expect(component).toBeTruthy();
        expect(listInstance).toBeTruthy();
    });

    it('should render all items', async(async()=> {
        const jsonData: DataModel[] = data.getAll;
        component.vehicles = jsonData;
        component.cdf.markForCheck();
        fixture.autoDetectChanges(true);
        await fixture.whenStable().then(async () => {
            expect(fixture).toMatchSnapshot();
        });
    }));

    it('should render only red items', async(async()=> {
        const redData: DataModel[] = redItems;
        component.vehicles = redData;
        component.cdf.markForCheck();
        fixture.autoDetectChanges(true);
        await fixture.whenStable().then(async () => {
            expect(fixture).toMatchSnapshot();
        });
    }));

});
