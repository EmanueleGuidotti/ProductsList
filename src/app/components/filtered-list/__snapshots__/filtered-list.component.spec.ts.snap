// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilteredListComponent should render all items 1`] = `
<filtered-list-component
  cdf={[Function ViewRef_]}
  vehicles={[Function Array]}
>
  <mat-grid-list
    class="mat-grid-list"
    cols="3"
    md-cols="3"
    md-gutter="1em"
    rowheight="100"
  >
    <div>
      
      <mat-grid-tile
        class="mat-grid-tile"
        style="left: 0px; top: 0px;"
      >
        <figure
          class="mat-figure"
        >
          <mat-card
            class="vehicle-item mat-card"
          >
            <img
              src="https://upload.wikimedia.org/wikipedia/commons/thumb/c/c9/Bugatti_Veyron_16.4_%E2%80%93_Frontansicht_%281%29%2C_5._April_2012%2C_D%C3%BCsseldorf.jpg/520px-Bugatti_Veyron_16.4_%E2%80%93_Frontansicht_%281%29%2C_5._April_2012%2C_D%C3%BCsseldorf.jpg"
            />
            <h4
              class="type"
            >
              car
            </h4>
            <p
              class="brand"
            >
              Bugatti Veyron
            </p>
          </mat-card>
        </figure>
      </mat-grid-tile>
      <mat-grid-tile
        class="mat-grid-tile"
        style="top: 0px;"
      >
        <figure
          class="mat-figure"
        >
          <mat-card
            class="vehicle-item mat-card"
          >
            <img
              src="https://upload.wikimedia.org/wikipedia/commons/thumb/1/15/All_Nippon_Airways_Boeing_787-8_Dreamliner_JA801A_OKJ_in_flight.jpg/600px-All_Nippon_Airways_Boeing_787-8_Dreamliner_JA801A_OKJ_in_flight.jpg"
            />
            <h4
              class="type"
            >
              airplane
            </h4>
            <p
              class="brand"
            >
              Boeing 787 Dreamliner
            </p>
          </mat-card>
        </figure>
      </mat-grid-tile>
      <mat-grid-tile
        class="mat-grid-tile"
        style="top: 0px;"
      >
        <figure
          class="mat-figure"
        >
          <mat-card
            class="vehicle-item mat-card"
          >
            <img
              src="https://upload.wikimedia.org/wikipedia/en/thumb/a/a1/UP_4466_Neil916.JPG/600px-UP_4466_Neil916.JPG"
            />
            <h4
              class="type"
            >
              train
            </h4>
            <p
              class="brand"
            >
              USRA 0-6-6
            </p>
          </mat-card>
        </figure>
      </mat-grid-tile>
      <mat-grid-tile
        class="mat-grid-tile"
        style="left: 0px;"
      >
        <figure
          class="mat-figure"
        >
          <mat-card
            class="vehicle-item mat-card"
          >
            <img
              src="https://upload.wikimedia.org/wikipedia/commons/thumb/1/1f/BOAC_C-4_Argonaut_Heathrow_1954.jpg/600px-BOAC_C-4_Argonaut_Heathrow_1954.jpg"
            />
            <h4
              class="type"
            >
              airplane
            </h4>
            <p
              class="brand"
            >
              Canadair North Star
            </p>
          </mat-card>
        </figure>
      </mat-grid-tile>
      <mat-grid-tile
        class="mat-grid-tile"
      >
        <figure
          class="mat-figure"
        >
          <mat-card
            class="vehicle-item mat-card"
          >
            <img
              src="https://upload.wikimedia.org/wikipedia/commons/thumb/1/1a/A400M-1969.jpg/600px-A400M-1969.jpg"
            />
            <h4
              class="type"
            >
              airplane
            </h4>
            <p
              class="brand"
            >
              Airbus A400M Atlas
            </p>
          </mat-card>
        </figure>
      </mat-grid-tile>
      <mat-grid-tile
        class="mat-grid-tile"
      >
        <figure
          class="mat-figure"
        >
          <mat-card
            class="vehicle-item mat-card"
          >
            <img
              src="https://upload.wikimedia.org/wikipedia/commons/e/e5/Bloch_MB_131_San_Diego_Air_%26_Space_Museum_3.jpg"
            />
            <h4
              class="type"
            >
              airplane
            </h4>
            <p
              class="brand"
            >
              Bloch MB.131
            </p>
          </mat-card>
        </figure>
      </mat-grid-tile>
      <mat-grid-tile
        class="mat-grid-tile"
        style="left: 0px;"
      >
        <figure
          class="mat-figure"
        >
          <mat-card
            class="vehicle-item mat-card"
          >
            <img
              src="https://upload.wikimedia.org/wikipedia/commons/thumb/d/d2/CFR_Steam_locomotive.jpg/600px-CFR_Steam_locomotive.jpg"
            />
            <h4
              class="type"
            >
              train
            </h4>
            <p
              class="brand"
            >
              Prairie 2-6-2
            </p>
          </mat-card>
        </figure>
      </mat-grid-tile>
      <mat-grid-tile
        class="mat-grid-tile"
      >
        <figure
          class="mat-figure"
        >
          <mat-card
            class="vehicle-item mat-card"
          >
            <img
              src="https://upload.wikimedia.org/wikipedia/commons/thumb/5/52/EMD_GP40_B%26M_339_Wells_Maine.jpg/600px-EMD_GP40_B%26M_339_Wells_Maine.jpg"
            />
            <h4
              class="type"
            >
              train
            </h4>
            <p
              class="brand"
            >
              EMD GP40
            </p>
          </mat-card>
        </figure>
      </mat-grid-tile>
      <mat-grid-tile
        class="mat-grid-tile"
      >
        <figure
          class="mat-figure"
        >
          <mat-card
            class="vehicle-item mat-card"
          >
            <img
              src="https://upload.wikimedia.org/wikipedia/commons/thumb/1/1c/440woodcut.jpg/600px-440woodcut.jpg"
            />
            <h4
              class="type"
            >
              train
            </h4>
            <p
              class="brand"
            >
              Amer 4-4-0
            </p>
          </mat-card>
        </figure>
      </mat-grid-tile>
      <mat-grid-tile
        class="mat-grid-tile"
        style="left: 0px;"
      >
        <figure
          class="mat-figure"
        >
          <mat-card
            class="vehicle-item mat-card"
          >
            <img
              src="https://upload.wikimedia.org/wikipedia/commons/thumb/c/cb/F40_Ferrari_20090509.jpg/1920px-F40_Ferrari_20090509.jpg"
            />
            <h4
              class="type"
            >
              car
            </h4>
            <p
              class="brand"
            >
              Ferrari F40
            </p>
          </mat-card>
        </figure>
      </mat-grid-tile>
      <mat-grid-tile
        class="mat-grid-tile"
      >
        <figure
          class="mat-figure"
        >
          <mat-card
            class="vehicle-item mat-card"
          >
            <img
              src="https://upload.wikimedia.org/wikipedia/commons/thumb/e/e5/2014-03-04_Geneva_Motor_Show_1379.JPG/440px-2014-03-04_Geneva_Motor_Show_1379.JPG"
            />
            <h4
              class="type"
            >
              car
            </h4>
            <p
              class="brand"
            >
              Lamborghini Hurac√°n
            </p>
          </mat-card>
        </figure>
      </mat-grid-tile>
      <mat-grid-tile
        class="mat-grid-tile"
      >
        <figure
          class="mat-figure"
        >
          <mat-card
            class="vehicle-item mat-card"
          >
            <img
              src="https://upload.wikimedia.org/wikipedia/commons/thumb/6/65/Porsche_Carrera_GT_-_Goodwood_Breakfast_Club_%28July_2008%29.jpg/440px-Porsche_Carrera_GT_-_Goodwood_Breakfast_Club_%28July_2008%29.jpg"
            />
            <h4
              class="type"
            >
              car
            </h4>
            <p
              class="brand"
            >
              Porsche Carrera GT
            </p>
          </mat-card>
        </figure>
      </mat-grid-tile>
    </div>
  </mat-grid-list>
</filtered-list-component>
`;

exports[`FilteredListComponent should render only red items 1`] = `
<filtered-list-component
  cdf={[Function ViewRef_]}
  vehicles={[Function Array]}
>
  <mat-grid-list
    class="mat-grid-list"
    cols="3"
    md-cols="3"
    md-gutter="1em"
    rowheight="100"
  >
    <div>
      
      <mat-grid-tile
        class="mat-grid-tile"
        style="left: 0px; top: 0px;"
      >
        <figure
          class="mat-figure"
        >
          <mat-card
            class="vehicle-item mat-card"
          >
            <img
              src="https://upload.wikimedia.org/wikipedia/commons/thumb/c/c9/Bugatti_Veyron_16.4_%E2%80%93_Frontansicht_%281%29%2C_5._April_2012%2C_D%C3%BCsseldorf.jpg/520px-Bugatti_Veyron_16.4_%E2%80%93_Frontansicht_%281%29%2C_5._April_2012%2C_D%C3%BCsseldorf.jpg"
            />
            <h4
              class="type"
            >
              car
            </h4>
            <p
              class="brand"
            >
              Bugatti Veyron
            </p>
          </mat-card>
        </figure>
      </mat-grid-tile>
      <mat-grid-tile
        class="mat-grid-tile"
        style="top: 0px;"
      >
        <figure
          class="mat-figure"
        >
          <mat-card
            class="vehicle-item mat-card"
          >
            <img
              src="https://upload.wikimedia.org/wikipedia/commons/thumb/1/15/All_Nippon_Airways_Boeing_787-8_Dreamliner_JA801A_OKJ_in_flight.jpg/600px-All_Nippon_Airways_Boeing_787-8_Dreamliner_JA801A_OKJ_in_flight.jpg"
            />
            <h4
              class="type"
            >
              airplane
            </h4>
            <p
              class="brand"
            >
              Boeing 787 Dreamliner
            </p>
          </mat-card>
        </figure>
      </mat-grid-tile>
      <mat-grid-tile
        class="mat-grid-tile"
        style="top: 0px;"
      >
        <figure
          class="mat-figure"
        >
          <mat-card
            class="vehicle-item mat-card"
          >
            <img
              src="https://upload.wikimedia.org/wikipedia/commons/thumb/1/1f/BOAC_C-4_Argonaut_Heathrow_1954.jpg/600px-BOAC_C-4_Argonaut_Heathrow_1954.jpg"
            />
            <h4
              class="type"
            >
              airplane
            </h4>
            <p
              class="brand"
            >
              Canadair North Star
            </p>
          </mat-card>
        </figure>
      </mat-grid-tile>
      <mat-grid-tile
        class="mat-grid-tile"
        style="left: 0px;"
      >
        <figure
          class="mat-figure"
        >
          <mat-card
            class="vehicle-item mat-card"
          >
            <img
              src="https://upload.wikimedia.org/wikipedia/commons/thumb/1/1a/A400M-1969.jpg/600px-A400M-1969.jpg"
            />
            <h4
              class="type"
            >
              airplane
            </h4>
            <p
              class="brand"
            >
              Airbus A400M Atlas
            </p>
          </mat-card>
        </figure>
      </mat-grid-tile>
      <mat-grid-tile
        class="mat-grid-tile"
      >
        <figure
          class="mat-figure"
        >
          <mat-card
            class="vehicle-item mat-card"
          >
            <img
              src="https://upload.wikimedia.org/wikipedia/commons/thumb/d/d2/CFR_Steam_locomotive.jpg/600px-CFR_Steam_locomotive.jpg"
            />
            <h4
              class="type"
            >
              train
            </h4>
            <p
              class="brand"
            >
              Prairie 2-6-2
            </p>
          </mat-card>
        </figure>
      </mat-grid-tile>
      <mat-grid-tile
        class="mat-grid-tile"
      >
        <figure
          class="mat-figure"
        >
          <mat-card
            class="vehicle-item mat-card"
          >
            <img
              src="https://upload.wikimedia.org/wikipedia/commons/thumb/1/1c/440woodcut.jpg/600px-440woodcut.jpg"
            />
            <h4
              class="type"
            >
              train
            </h4>
            <p
              class="brand"
            >
              Amer 4-4-0
            </p>
          </mat-card>
        </figure>
      </mat-grid-tile>
      <mat-grid-tile
        class="mat-grid-tile"
        style="left: 0px;"
      >
        <figure
          class="mat-figure"
        >
          <mat-card
            class="vehicle-item mat-card"
          >
            <img
              src="https://upload.wikimedia.org/wikipedia/commons/thumb/c/cb/F40_Ferrari_20090509.jpg/1920px-F40_Ferrari_20090509.jpg"
            />
            <h4
              class="type"
            >
              car
            </h4>
            <p
              class="brand"
            >
              Ferrari F40
            </p>
          </mat-card>
        </figure>
      </mat-grid-tile>
    </div>
  </mat-grid-list>
</filtered-list-component>
`;
